RetailCoder.VBE/UI/Inspections/InspectionResultsViewModel.cs

```diff
@@ -102,7 +102,7 @@ private void _configService_SettingsChanged(object sender, ConfigurationChangedE
         {            
             if (e.InspectionSettingsChanged)
             {
-                RefreshInspections();
+                RefreshInspections(CancellationToken.None);
```

Comment:
I do not see any suitable tokens at hand here. However, I could imagine requesting a reparse like when refreshing or is that overkill. 
On the other hand, do we really need to refresh after changing the setting? I would not find it to be surprising if the inspections only changed on the next regular refresh.

Reply:
If the inspections can be run async (I'm not sure if that is the case here; I assume not since it's missing the `Async` suffix here), it needs to be cancellable, or we will risk COM access violations when a COM event gets fired that requires removal. In that case, we'd need a class to dispose a cancellation token to be used by all consuming code.

If this is all synchronous, then I think we are OK to not worry about providing a token.

Reply:
This is in no way synchronous. Calling an async method without awaiting basically means that you follow it to its first await and then abandon it. You implicitly throw away the task returned by the method.

Reply:
If it is in fact async, then I really think the name should have that `Async` suffix to remind us of that fact, even though we are not awaiting. 

It also tells me that we really need a token here. Which means you need something to dispense a token for all classes to use because you can't know when a COM event that removes something gets fired warranting a cancellation of that manual refresh.

Reply:
I'm pretty sure that we can just call for a Reparse here. The parsing should be either really quick (because nothing apart from the settings changed) or it was already necessary, so no harm done.

Reply:
Now, I am confused. The enclosing method (an event handler) is not asnynchronous. It is like a method firing up a background task with `Task.Run` for some action and then forgetting about the task. 

Creating something you envision here, is really not in the scope of this PR.
For this PR, I see three options:
1) Leave it as it is.
2) Request a reparse on configuration changes followed by the inspection run like for the manual refresh. (This triggers the inspections from the state change to `Ready`, i.e. with the token from the parse.)
3) Remove rerunning the inspections on configuration changes altogether.

Reply:
Never mind; it's me who is confused. I think #2, as @Vogel612  suggested is the best of the option.